### DISCUSSION TITLE
Question: Heating mug in morning automatically?

### DISCUSSION THREAD

**Discussion by jgdf42** [sopelj/hass-ember-mug-component#112] [Announcements]
I wish this mug could begin heating at certain times. Is there any way to exploit the api to make this happen? It's my understanding the mug only begins heating when lifted. Would using a smartplug to kill power to the charger in the morning initiate the heating element?

**Comment by dshokouhi** [comment]
> It's my understanding the mug only begins heating when lifted. 

In my own experience it begins heating as soon as it detected liquid.

**Comment by jgdf42** [comment]
> > It's my understanding the mug only begins heating when lifted.
> 
> In my own experience it begins heating as soon as it detected liquid.

But I want to fill the mug at night and wake up to it hot in the morning.

**Comment by sopelj** [comment]
That's an interesting idea. I'm not sure if it would work, but you could try turning temperature control off, filling it and then turning it back on when you want it to heat up, but I'm not sure if it would work. It requires a lot more energy to heat a liquid from cold than it does to keep it warm. It would definitely have to be on the stand. If that works, then you could add an automation in Home Assistant to do it when you want, either at a specific time of based on an event.

**Comment by tomchambers2** [comment]
Seems like this might work, however the issue I think you'd run into is that the mug uses significant temperature differences to measure whether it's full and if it's say 15-20C, the detection becomes very flaky and if it reads as empty, it's not going to turn on.

However if there's a way of getting round that seems like it would work. I am currently heating up from 20C to 60C, takes a while though, about 30 minutes and it's only half full. It's not a kettle at the end of the day!

https://support.ember.com/hc/en-us/articles/360048983131-Why-is-my-Ember-reading-as-empty
